ifrs17/Utils/Queries.ipynb

```diff
@@ -217,14 +217,26 @@
         {
             "cell_type": "code",
             "source": [
-                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args, ",
-                "\n                                                                                                      IEnumerable<DataNodeData> dataNodes)",
-                "\n{",
-                "\n    var contractualCurrenciesInScope = dataNodes.Select(dn => dn.ContractualCurrency).ToHashSet();",
-                "\n    return (await querySource.LoadCurrentAndPreviousParameterAsync<YieldCurve>(args, ",
+                "public static async Task<Dictionary<string, Dictionary<int, YieldCurve>>> LoadCurrentYieldCurveAsync(this IQuerySource querySource, Args args,",
+                "\n                                                                                                                                IEnumerable<DataNodeData> dataNodes)",
+                "\n{    ",
+                "\n    var currentYieldCurveByGoc = new Dictionary<string,  Dictionary<int, YieldCurve>>();",
+                "\n    ",
+                "\n    foreach (var dn in dataNodes)",
```

Comment:
The current curve is not DN dependent, right? Why do you need this at all?

Reply:
Sara: The current curve is dn dependendent beacuse of YieldCurveName

Reply:
But then you should extract the name, no? You are calling the query for every DN, even if the name is null for all. The foreach should be on the distinct names + the null.

Reply:
I understand what you mean. And I think it should work. One has that to adapt the getter in the storage accordingly. 
If we put the YCName as key of the dict we can define and use a defaultKey = "__D" in const.

Reply:
implemented